app/server.R

```diff
@@ -908,9 +912,9 @@ server <- function(input, output, session) {
     # Make sure we have a valid as of selection
     nonValidAsOf <- selectedAsOf == "" || !(as.Date(selectedAsOf) %in% asOfChoices)
     if (length(asOfChoices) != 0 && nonValidAsOf) {
-      selectedAsOf <- max(asOfChoices, na.rm = TRUE)
+      selectedAsOf <- max(asOfChoices)
     }
-    AS_OF_CHOICES(sort(unique(asOfChoices)))
+    AS_OF_CHOICES(sort(asOfChoices))
```

Comment:
question: is `sort` better here or on line 894?

it looks like you moved `unique` to `distinct` where `asOfChoices` is declared; does it make sense to do the ordering at declaration time too or no?

Reply:
I considered moving the `sort` up to line 894, but I'm leaning towards not (but open to arguments in either direction). It would be easy to change the logic upstream of the place where we actually render the as-of choices in the sidebar so that they're not sorted anymore, so I think it's a bit safer to sort right before displaying.

On the other hand, removing duplicates farther in advance should save some time/resources downstream.